### PR TITLE
Add pages + functions

### PR DESCRIPTION
**Pull request by Albertozhao**

### CODE REVIEW

File: products/pages/src/content/platform/functions.md [hunk 1]
```diff
@@ -0,0 +1,366 @@
+---
+order: 6
+pcx-content-type: concept
+---
+
+# Functions
+With Pages, you can now build full-stack applications by executing code on the Cloudflare network with help from [Cloudflare Workers](https://workers.cloudflare.com/). Functions enable you to run server-side code to enable dynamic functionality without running a dedicated server. With Functions, you can introduce application aspects  such as authenticating, querying databases, handling form submissions, or working with middleware.
```
Comment (deadlypants1973) [inline products/pages/src/content/platform/functions.md]:
```suggestion
# Functions

With Pages, you can now build full-stack applications by executing code on the Cloudflare network with help from [Cloudflare Workers](https://workers.cloudflare.com/). Functions enable you to run server-side code to enable dynamic functionality without running a dedicated server. With Functions, you can introduce application aspects such as authenticating, querying databases, handling form submissions, or working with middleware.
```

File: products/pages/src/content/platform/functions.md [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+ 
+## Built with Cloudflare Workers
+ 
+Cloudflare Workers provides a serverless [execution environment](https://www.cloudflare.com/en-gb/learning/serverless/what-is-serverless/) that allows you to create entirely new applications or augment existing ones without configuring or maintaining infrastructure. 
+ 
+Previously, you could only add dynamic functionality to your Pages site by manually deploying a Worker using Wrangler, which meant that your application is written across both Pages and Workers. Functions allow you to leverage the Workers platform directly from within a Pages project by utilizing a project’s filesystem convention. This enables you to deploy your entire site – both its static and dynamic content – when you `git push`. 
+ 
+## Setup
+ 
+To get started, you’ll need to create a `/functions` directory at the root of your project. Writing your functions files in this directory will automatically generate a Worker with custom functionality at the predesignated routes.
```
Comment (deadlypants1973) [inline products/pages/src/content/platform/functions.md]:
```suggestion
To get started, create a `/functions` directory at the root of your project. Writing your functions files in this directory will automatically generate a Worker with custom functionality at the predesignated routes.
```

File: products/pages/src/content/platform/functions.md [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+ 
+## Functions routing
+Using a `/functions` directory will generate a routing table based on the files present in the directory. You may use JavaScript (`*.js`) or TypeScript (`*.ts`) to write your Functions. We declare a `PagesFunction` type in the [@cloudflare/workers-types](https://github.com/cloudflare/workers-types) library which you can use to type-check your Functions.
```
Comment (deadlypants1973) [inline products/pages/src/content/platform/functions.md]:
```suggestion
## Functions routing

Using a `/functions` directory will generate a routing table based on the files present in the directory. You may use JavaScript (`*.js`) or TypeScript (`*.ts`) to write your Functions. A `PagesFunction` type is declared in the [@cloudflare/workers-types](https://github.com/cloudflare/workers-types) library which you can use to type-check your Functions. 
```

File: products/pages/src/content/platform/functions.md [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+/api/time => ./functions/time.ts
+/api/todos => ./functions/todos/index.ts
+/api/todos/* => ./functions/todos/[id].ts
+/api/todos/*/** => ./functions/todos/[[path]].ts
+/*/profile => ./functions/[username]/profile.ts
+/** => ./functions/[[path]].ts
+```
+ 
+### Path segments
+Note that in the example above:
```
Comment (deadlypants1973) [inline products/pages/src/content/platform/functions.md]:
```suggestion
### Path segments

Note that in the example above:
```

File: products/pages/src/content/platform/functions.md [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+* A `*` denotes a placeholder for a single path segment (e.g. /todos/123)
```
Comment (deadlypants1973) [inline products/pages/src/content/platform/functions.md]:
```suggestion
* A `*` denotes a placeholder for a single path segment (for example, `/todos/123`)
```

File: products/pages/src/content/platform/functions.md [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+* A `**` matches one or more path segments (e.g. /todos/123/dates/confirm).
```
Comment (deadlypants1973) [inline products/pages/src/content/platform/functions.md]:
```suggestion
* A `**` matches one or more path segments (for example, `/todos/123/dates/confirm`). 
```

File: products/pages/src/content/platform/functions.md [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+When naming your files:
+* `[name]` is a placeholder for a single path segment
+* `[[name]]` matches any depth of route below this point
+
+When a filename includes a placeholder, the `name` must be alphanumeric and cannot contain spaces. In turn, the URL segment(s) that match the placeholder will be available under the `context.params` object using the filename placeholder as the key.
+
+<Aside type=”note” header=”Route Specificity”> 
+
+More specific routes (i.e. those with fewer wildcards) take precedence over less specific routes.
+</Aside>
```
Comment (deadlypants1973) [inline products/pages/src/content/platform/functions.md]:
```suggestion
More specific routes (that is, those with fewer wildcards) take precedence over less specific routes.

</Aside>
```

**Review by deadlypants1973 — APPROVED** [review]

**Review by kristianfreeman — APPROVED** [review]

File: products/pages/src/content/platform/functions.md [hunk 8]
```diff
@@ -0,0 +1,370 @@
...
+
+
+
+### Durable Object namespace
+
+Durable Objects are Cloudflare’s strongly consistent coordination primitive that makes connecting WebSockets, handling state, and building entire applications a breeze. As with Workers KV, you first have to [create the Durable Object](https://developers.cloudflare.com/workers/learning/using-durable-objects#uploading-a-durable-object-worker) and then you can configure it as a binding to your Pages project. Navigate to the Pages interface in **Settings** > **Functions** to select the namespace and set a name. 
+
+![DO-Binding](DO-functions.png)
+
+### Environment variable
```
Comment (WalshyDev) [inline products/pages/src/content/platform/functions.md:292]: I'd maybe note that secrets cannot be set currently.

**Review by WalshyDev — CHANGES_REQUESTED** [review]

**Pull request merged**